adapters/networkHandlerFactory.js

```diff
@@ -1,26 +1,38 @@
 const {
-  networkHandler: BrazeNetworkHandler
-} = require("../v0/destinations/braze/util");
-const {
-  networkHandler: MarketoNetworkHandler
-} = require("../v0/destinations/marketo/util");
-const {
-  networkHandler: GenericNetworkHandler
+  GenericNetworkHandler
 } = require("./networkhandler/genericNetworkHandler");
-const {
-  networkHandler: BqStreamNetworkHandler
-} = require("../v0/destinations/bqstream/util");
+const NetWorkHandlerClassMap = require("./networkHandlersMap");
+
+class NetworkHandlerFactory {
+  constructor() {
+    this.info = "NetworkHandlerFactory";
+    // intially map would contain only the generic handler
+    this.factoryMap = new Map();
+    this.factoryMap.set("generic", new GenericNetworkHandler());
```

Comment:
We should avoid creating generic handler by default (make it on-demand)

Reply:
IMHO the map should at least start off with the generic handler. Anyway that is the main one that would be used. Only for few specific destinations we are going to have a specific/custom handler

Reply:
Got it. But why create if we don't know for sure?